src/java.base/share/classes/jdk/internal/foreign/abi/AbstractLinker.java

```diff
@@ -121,4 +134,30 @@ private static void checkHasNaturalAlignment(MemoryLayout layout) {
             throw new IllegalArgumentException("Layout bit alignment must be natural alignment: " + layout);
         }
     }
+
+    private static MemoryLayout stripNames(MemoryLayout ml) {
+        // we don't care about transferring alignment and byte order here
+        // since the linker already restricts those such that they will always be the same
+        return switch (ml) {
+            case StructLayout sl -> MemoryLayout.structLayout(stripNames(sl.memberLayouts()));
+            case UnionLayout ul -> MemoryLayout.unionLayout(stripNames(ul.memberLayouts()));
+            case SequenceLayout sl -> MemoryLayout.sequenceLayout(sl.elementCount(), stripNames(sl.elementLayout()));
+            case AddressLayout al -> al.targetLayout()
+                    .map(tl -> al.withoutName().withTargetLayout(stripNames(tl)))
+                    .orElse(al.withoutName());
```

Comment:
This can use [`Optional::orElseGet`]:
```suggestion
                    .orElseGet(al::withoutName);
```

[`Optional::orElseGet`]: https://docs.oracle.com/en/java/javase/20/docs/api/java.base/java/util/Optional.html#orElseGet%28java.util.function.Supplier%29